test/src/upgrade_eks_test.go

```diff
@@ -0,0 +1,230 @@
+
+	k8s.WaitUntilServiceAvailable(suite.T(), kubeCtlOptions, "whoami-service", 60, 1*time.Second)
+	// wait to ensure service available
+	time.Sleep(30 * time.Second)
+
+	// Now we verify that the service will successfully boot and start serving requests
+	localPort1 := 8883
+
+	service := k8s.GetService(suite.T(), kubeCtlOptions, "whoami-service")
+	portForwardProc1 := k8s.NewTunnel(kubeCtlOptions, k8s.ResourceTypeService, service.ObjectMeta.Name, localPort1, 80)
+	defer portForwardProc1.Close()
+	portForwardProc1.ForwardPort(suite.T())
+
+	// wait for the port forward to be ready
+	time.Sleep(5 * time.Second)
+
+	http_helper.HttpGetWithRetryWithCustomValidation(
+		suite.T(),
+		fmt.Sprintf("http://%s", portForwardProc1.Endpoint()),
+		nil,
+		30,
+		10*time.Second,
+		func(statusCode int, body string) bool {
+			return statusCode == 200
+		},
+	)
+
+	// upgrade the cluster
+	suite.varTf["kubernetes_version"] = "1.29"
+	suite.sugaredLogger.Infow(fmt.Sprintf("Upgrading the EKS cluster to v%s using aws sdk", suite.varTf["kubernetes_version"]), "extraVars", suite.varTf)
+	errUpdate := utils.UpgradeEKS(context.Background(), eksSvc, suite.clusterName, suite.varTf["kubernetes_version"].(string))
+	suite.Require().NoError(errUpdate)
+
+	suite.sugaredLogger.Infow("Waiting for worker nodes to join the EKS cluster after the upgrade")
+	errClusterReady = utils.WaitUntilKubeClusterIsReady(result.Cluster, 5*time.Minute, uint64(suite.expectedNodes))
+	suite.Require().NoError(errClusterReady)
+
+	// perform update with terraform
+	terraformOptions = &terraform.Options{
+		TerraformDir: tfDir,
+		Upgrade:      false,
+		VarFiles:     []string{"../fixtures/fixtures.default.eks.tfvars"},
+		Vars:         suite.varTf,
+	}
+
+	suite.sugaredLogger.Infow("Reapply terraform after EKS cluster upgrade...", "extraVars", suite.varTf)
+
+	if cleanClusterAtTheEnd == "true" {
+		defer terraform.Destroy(suite.T(), terraformOptions)
+		defer runtime.HandleCrash(func(i interface{}) {
+			terraform.Destroy(suite.T(), terraformOptions)
+		})
+	}
+
+	terraform.InitAndApplyAndIdempotent(suite.T(), terraformOptions)
+
+	errClusterReady = utils.WaitUntilKubeClusterIsReady(result.Cluster, 5*time.Minute, uint64(suite.expectedNodes))
+	suite.Require().NoError(errClusterReady)
+
+	// Check version of the upgraded cluster
+	result, err = eksSvc.DescribeCluster(context.Background(), inputEKS)
+	suite.Assert().NoError(err)
+	suite.Assert().Equal(suite.varTf["kubernetes_version"], *result.Cluster.Version)
+
+	// check everything works as expected
+	k8s.WaitUntilServiceAvailable(suite.T(), kubeCtlOptions, "whoami-service", 60, 1*time.Second)
+	// wait to ensure service available
+	time.Sleep(30 * time.Second)
+
+	// Forward port again
+	localPort2 := 8887
+	service = k8s.GetService(suite.T(), kubeCtlOptions, "whoami-service")
+	portForwardProc2 := k8s.NewTunnel(kubeCtlOptions, k8s.ResourceTypeService, service.ObjectMeta.Name, localPort2, 80)
```

Comment:
same mentioned as above on setting it simply to `0` to leave the task to terratest on finding an available port.